readLog/src/main/java/org/corfudb/readLog/readLog.java

```diff
@@ -0,0 +1,354 @@
+        reader.cleanUp();
+    }
+
+    public readLog() {
+        fileStreamIn = null;
+        fileStreamOut = null;
+    }
+
+    public int run(String[] args) {
+        try {
+            boolean ret = init(args);
+        } catch(DocoptExitException e) {
+            System.out.println(USAGE);
+            System.exit(1);
+        }
+        return readAll();
+    }
+
+    public boolean init(String[] args) {
+        Docopt parser = new Docopt(USAGE);
+        parser.withExit(false);
+        Map<String, Object> opts = parser.parse(args);
+        op = new Operation(Operation.OperationType.REPORT, 0);
+        String log_file_name = new String();
+        boolean use_output_file = false;
+        if (opts.get("<log_file>") != null) {
+            log_file_name = (String) opts.get("<log_file>");
+            System.out.println("Log file: " + log_file_name);
+        } else {
+            System.out.print(USAGE);
+            System.out.println("Return FALSE");
+            return false;
+        }
+        if (opts.get("--erase") != null) {
+            int addr = Integer.parseInt(((String) opts.get("--erase")).trim());
+            op = new Operation(Operation.OperationType.ERASE_ONE, addr);
+            use_output_file = true;
+        } else if (opts.get("--erase_tail") != null) {
+            int addr = Integer.parseInt(((String) opts.get("--erase_tail")).trim());
+            op = new Operation(Operation.OperationType.ERASE_TAIL, addr);
+            use_output_file = true;
+        } else if (opts.get("--display") != null) {
+            int flag = Integer.parseInt(((String) opts.get("--display")).trim());
+            if (flag == 0) {
+                op = new Operation(Operation.OperationType.DISPLAY, 0);
+            } else {
+                op = new Operation(Operation.OperationType.DISPLAY_ALL, 0);
+            }
+        }
+        LogHeader lh = LogHeader.newBuilder()
+                .setVersion(10)
+                .setVerifyChecksum(true)
+                .build();
+        LogHeaderSize = lh.getSerializedSize();
+        Metadata md = Metadata.newBuilder()
+                .setChecksum(0x4d5ca4cc)
+                .setLength(lh.getSerializedSize())
+                .build();
+        MetadataSize = md.getSerializedSize();
+
+        File f_in = new File(log_file_name);
+        File f_out = use_output_file ? new File(log_file_name + ".modified") : null;
+        if (f_in.canRead()) {
+            try {
+                fileStreamIn = new FileInputStream(f_in);
+                fileChannelIn = fileStreamIn.getChannel();
+                if (use_output_file) {
+                    fileStreamOut = new FileOutputStream(f_out);
+                    fileChannelOut = fileStreamOut.getChannel();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return true;
+        }
+        return false;
+    }
+
+    int readAll() {
+        int record_cnt = 0;
```

Comment:
As of this version it only counts the records in a file, so it won't overflow.  If it were reading the whole directory a single counter might not be the right / only thing to track.